generic/SmoothL1Criterion.c

```diff
@@ -0,0 +1,60 @@
+#ifndef TH_GENERIC_FILE
+#define TH_GENERIC_FILE "generic/SmoothL1Criterion.c"
+#else
+
+static int nn_(SmoothL1Criterion_updateOutput)(lua_State *L)
+{
+  THTensor *input = luaT_checkudata(L, 2, torch_Tensor);
+  THTensor *target = luaT_checkudata(L, 3, torch_Tensor);
+  int sizeAverage = luaT_getfieldcheckboolean(L, 1, "sizeAverage");
+  real sum;
+
+  sum = 0;
+  TH_TENSOR_APPLY2(real, input, real, target,
+                   real z = fabs(*input_data - *target_data);
```

Comment:
Isn't torch itself using fabs for FloatTensors as well? see https://github.com/torch/torch7/blob/086ec2a760a15a5165b488e25899f3ee70e2ec6e/TensorMath.lua#L946